app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,489 @@
+   */
+  handleEditSubmit = updatedCloseContactData => {
+    const currentCloseContactId = this.state.table.rowData[this.state.editRow]?.id;
+    updatedCloseContactData['id'] = currentCloseContactId;
+    this.setState(
+      {
+        showEditModal: false,
+        editRow: null,
+      },
+      () => {
+        this.updateCloseContact(updatedCloseContactData, true);
+      }
+    );
+  };
+
+  // Toggles the delete modal and clears any reason or reason text
+  toggleDeleteModal = id => {
+    let selectedCloseContactID = _.isNumber(id) ? id : null;
+    this.setState({
+      showDeleteModal: !this.state.showDeleteModal,
+      selectedCloseContactID,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleDeleteSubmit = () => {
+    let delete_reason = this.state.delete_reason;
+    if (delete_reason === 'Other' && this.state.delete_reason_text) {
+      delete_reason += ', ' + this.state.delete_reason_text;
+    }
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(`${window.BASE_PATH}/close_contacts/${this.state.selectedCloseContactID}`, {
+          data: {
+            patient_id: this.props.patient.id,
+            delete_reason,
+          },
+        })
+        .then(() => {
+          location.reload();
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  handleCCDeleteUpdate = event => {
+    if (['delete_reason', 'delete_reason_text'].includes(event?.target?.id)) {
+      this.setState({ [event.target.id]: event?.target?.value });
+    }
+  };
+
+  /**
+   * Makes a request to update an existing close contact record on the backend and reloads page once complete.
+   * @param {*} newCloseContactData
+   * @param {*} isEdit - whether this is an edit or a new close contact
+   */
+  updateCloseContact = (ccData, isEdit) => {
+    this.setState({ isLoading: true }, () => {
+      let url = `${window.BASE_PATH}/close_contacts`;
+      if (isEdit) {
+        url += `/${ccData.id}`;
+      }
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(url, {
+          patient_id: this.props.patient.id,
+          first_name: ccData.first_name || '',
+          last_name: ccData.last_name || '',
+          primary_telephone: ccData.primary_telephone ? formatPhoneNumber(ccData.primary_telephone) : '',
+          email: ccData.email || '',
+          last_date_of_exposure: ccData.last_date_of_exposure || null,
+          assigned_user: ccData.assigned_user || null,
+          notes: ccData.notes || '',
+          enrolled_id: ccData.enrolled_id || null,
+          contact_attempts: ccData.contact_attempts || 0,
+        })
+        .then(() => {
+          this.setState({ isLoading: false });
```

Comment:
Also necessary

Reply:
I think it should remain, because it is logically the correct action. We were loading, then the action finished, so we no-longer are. What if we change the page in the future to update without needing a reload? I dont think we should remove logically correct code, just because the page reloads.

@tstrass @holmesie I'm curious what your opinions are

Reply:
I totally get where you're coming from, but it's currently unnecessary and it is how we do it in other components.  If we do change that functionality down the road, we'd be updating this everywhere so adding it here wouldn't be a big workflow difference

Reply:
@tstrass curious your thoughts on this

Reply:
We should remove it. On top being unnecessary, It's not actually logically correct, because the data in the table doesn't reflect the updates until after the reload. In fact, visually it's better to have the spinner until the reload occurs, because otherwise, there is no indication to the user that something is going on in the background in the small but not insignificant time between when the modal is dismissed and when the page reload begins.